package.json

```diff
@@ -8,13 +8,19 @@
   "types": "dist/index.d.ts",
   "main": "dist/haws.umd.js",
   "module": "dist/index.js",
+  "exports": {
+    "node": {
+      "import": "./dist/index.js",
+      "require": "./dist/haws.cjs"
```

Comment:
Why don't we use the `UMD` build here?

Reply:
Any compatible format would work. The file extension is the thing that needs to be different so node stops treating it as an ES module.

@balloob said you had a possible solution to use two `package.json` which I also did get to work but was unsure what was the best approach on the directory structure.

Reply:
The `exports` would be best, we didn't go that route because it caused problems before, might be solved by now... Would be nice if we can find a changelog somewhere that fixed it.

I don't think the extension is needed anymore with `exports`, so we could just use `dist/haws.umd.js`.

Reply:
The directory structure with a second `package.json` would be an issue indeed, as it would be a breaking change.

Reply:
No, the error about it being an ES Module is thrown if you use `./dist/haws.umd.js` in place of `./dist/haws.cjs`

Reply:
> Would be nice if we can find a changelog somewhere that fixed it.

The previous error reported [here](https://github.com/home-assistant/home-assistant-js-websocket/issues/113#issuecomment-606005122). Can be replicated by using `dist/haws.cjs` instead of `./dist/haws.cjs`

```
Error [ERR_INVALID_PACKAGE_TARGET]: Invalid "exports" main target "dist/haws.cjs" defined in the package config /node-red-contrib-home-assistant-websocket/node_modules/home-assistant-js-websocket\package.json; targets must start with "./"
```

Reply:
> No, the error about it being an ES Module is thrown if you use `./dist/haws.umd.js` in place of `./dist/haws.cjs`

That makes no sense, `require` can only be CommonJS but ok... Node... :-/